libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/OAuthWebviewHelper.kt

```diff
@@ -738,7 +752,7 @@ open class OAuthWebviewHelper : KeyChainAliasCallback {
                 create(loginOptions.loginUrl),
                 loginOptions.oauthClientId,
                 code,
-                codeVerifier,
+                frontDoorBridgeCodeVerifier ?: codeVerifier,
```

Comment:
This just feels dangerous.  What if you login with qrCode (so `frontDoorBridgeCodeVerifier` is set), we logout and try a standard login?  Is `OAuthWebviewHelper` (and thus `frontDoorBridgeCodeVerifier`) still in memory?

Reply:
I was really curious about state here also.  My tests showed the login state and all the objects I expected being recreated each log in attempt.  That would match my expectation since a log out really should set the app back to a clean slate.

Additionally, I also just tested using a switch user/add user scenario while another user is logged in.  I got a new OAuthWebViewHelper instance, so everything is back to a clean log in state.

Do we have any scenarios where we actually would want to retain these objects?  I suspect that's a "no".

Reply:
Here's a thought that I was already considering: I feel we need handling when, somehow, a QR Code Log In fails. The user could theoretically fall back to username and password with the same web view.

In `OAuthWebViewHelper.onAuthFlowError`, we could null these two parameters.

Would it make sense to do the same in `onAuthFlowComplete`?  I see that is called _after_ `onWebServerFlowComplete` and `doCodeExchangeEndpoint`.

With both in place, that would help us reset state for error flows (where I think we need it) and on success flows (where we're discarding the `OAuthWebViewHelper` anyways).

I'm trying to do some destructive testing with failing QR Codes 💣

Reply:
Yeah, if QR Login somehow fails was the other scenario I was thinking of.  Might be good to reset it in `onAuthFlowError`.

Reply:
@brandonpage - Take a look at 👉🏻 https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/2594/commits/a2a35754b194e30abaccb51ac1373cafbd706e47 - That adds a new reset method and calls to it from three points in the logic where I reckon we need to get back to the initial state.  Of course, I'm not aware of a way to re-use the object after a successful authentication transaction.  That one may not be needed.

Reply:
The failure case is the most notable addition.